api/internal/localizer/locloader.go

```diff
@@ -0,0 +1,351 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/internal/git"
+	"sigs.k8s.io/kustomize/api/loader"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+const urlSeparator = "/"
+const replacement = "-"
+
+const dstPrefix = "localized"
+const localizeDir = dstPrefix + replacement + "files"
```

Comment:
This is a really complicated way to define things, it is easier to understand something like

```
const localizedFilesDir = "localized-files"
```

and just use "/" and "-" where needed. I think having consts for these are making it more confusing than it is helping.